### PR TITLE
build against dunai-0.6.0, bump version #131

### PR DESCRIPTION
**Pull request by smunix**

### CODE REVIEW

File: rhine/dunai.nix [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{ mkDerivation, base, MonadRandom, simple-affine-space, stdenv
```
Comment (turion) [inline rhine/dunai.nix]: Shouldn't we be able to use `dunai` from `stackage`/`nixpkgs`?

File: rhine/rhine.release.nix [hunk 2]
```diff
@@ -0,0 +1,17 @@
+let
+  config = {
+    packageOverrides = nixpkgs: rec {
+      haskellPackages = nixpkgs.haskellPackages.override {
+        overrides = hnew: hold: rec {
+          rhine = hnew.callPackage ./rhine.nix {};
```
Comment (turion) [inline rhine/rhine.release.nix]: Can we also use `callCabal2Nix` here?

File: rhine/src/FRP/Rhine/ClSF/Util.hs [hunk 3]
```diff
@@ -221,9 +223,9 @@ weightedAverageFrom v0 = feedback v0 $ proc ((v, weight), vAvg) -> do
 --   all features below a given time constant @t@.
 --   (Equivalently, it filters out frequencies above @1 / (2 * pi * t)@.)
 averageFrom
-  :: ( Monad m, VectorSpace v
-     , Floating (Groundfield v)
-     , Groundfield v ~ Diff td)
+  :: ( Monad m, VectorSpace v s
+     , Floating (s)
```
Comment (turion) [inline rhine/src/FRP/Rhine/ClSF/Util.hs]: I think you can elide these brackets here (and in some other similar places)

File: rhine/src/FRP/Rhine/Clock/Periodic.hs [hunk 4]
```diff
@@ -45,7 +45,7 @@ instance (Monad m, NonemptyNatList v)
   type Time (Periodic v) = Integer
   type Tag  (Periodic v) = ()
   initClock cl = return
-    ( cycleS (theList cl) >>> withSideEffect wait >>> sumS &&& arr (const ())
+    ( cycleS (theList cl) >>> withSideEffect wait >>> (accumulateWith (+) 0) &&& arr (const ())
```
Comment (turion) [inline rhine/src/FRP/Rhine/Clock/Periodic.hs]: `sumS` still exists in `dunai` afaikt, you have to `import Data.MonadicStreamFunction.Util`.

File: rhine-examples/rhine-examples.nix [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{ mkDerivation, base, random, rhine, stdenv, transformers }:
```
Comment (turion) [inline rhine-examples/rhine-examples.nix]: Same here. Does `callCabal2Nix` work as well?

File: rhine/src/FRP/Rhine/Clock/Periodic.hs (hunk 4 above)
Comment (smunix) [reply rhine/src/FRP/Rhine/Clock/Periodic.hs]: True, but the new sumS implementation requires instances of The VectorSpace. Integer isn't a VectorSpace.

Comment (turion) [reply rhine/src/FRP/Rhine/Clock/Periodic.hs]: Fair enough.

**Comment by smunix** [comment]
> In particular, thank you for the `nix` rig! Is it possible to add it to `.travis.yml`? I'm not sure one can CI both cabal and nix on travis.
> 
Sure, let me take a look.

**Comment by smunix** [comment]
how do you reopen this PR ? I closed it by mistake and I can't add more updates to it.

**Comment by turion** [comment]
There you go :)

**Comment by turion** [comment]
I just realised this is ready to go! Thanks a lot @smunix !

**Pull request merged**